### PR TITLE
Experiment UI includes and passes CCC to backend

### PR DESCRIPTION
**Pull request by skymandr**
Passing CCC information along from experiment -> job -> compile -> analysis.

- Could probably do with more tests. Help with this would be appreciated.
- I have not taken care of lint complaints to any great extent. I  am planning on doing this in a separate PR for the affected files once this is accepted.

### CODE REVIEW

File: scanomatic/ui_server_data/js/analysis.js [hunk 1]
```diff
@@ -214,7 +214,7 @@ function Analyse(button) {
             compilation: $("#compilation").val(),
             compile_instructions: $("#compile-instructions").val(),
             output_directory: $("#analysis-directory").val(),
-            ccc: $('#ccc-selection').val(),
+            cell_count_calibration: $('#ccc-selection').val(),
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/analysis.js]: Should be `cell_count_calibration_id`, should also be fixed for `/analysis` in `ui_server.py` and its html.

File: scanomatic/models/factories/scanning_factory.py [hunk 2]
```diff
@@ -210,6 +212,8 @@ class ScanningModelFactory(AbstractModelFactory):
         'computer': str,
         'version': str,
         'id': str,
+        'cell_count_calibration': (tuple, float),
```
Comment (local-minimum) [inline scanomatic/models/factories/scanning_factory.py]: could skip this, just need the `id` for the steps up to `analysis`-step

File: scanomatic/models/factories/scanning_factory.py [hunk 3]
```diff
@@ -220,9 +224,18 @@ class ScanningModelFactory(AbstractModelFactory):
...
 
-
         :rtype : scanomatic.models.scanning_model.ScanningModel
         """
+        if (not settings.get('cell_count_calibration_id', None) and
+                not settings.get('cell_count_calibration', None)):
+
+            settings['cell_count_calibration_id'] = 'default'
+
+        if (not settings.get('cell_count_calibration', None)):
```
Comment (local-minimum) [inline scanomatic/models/factories/scanning_factory.py]: see above

File: scanomatic/models/scanning_model.py [hunk 4]
```diff
@@ -104,7 +104,9 @@ def __init__(
             version=scanomatic.__version__,
             scanning_program="",
             scanning_program_version="",
-            scanning_program_params=tuple()):
+            scanning_program_params=tuple(),
+            cell_count_calibration=None,
```
Comment (local-minimum) [inline scanomatic/models/scanning_model.py]: remove, only keep `id`

File: scanomatic/models/scanning_model.py [hunk 5]
```diff
@@ -125,6 +127,8 @@ def __init__(
         self.mode = mode
         self.computer = computer
         self.start_time = start_time
+        self.cell_count_calibration = cell_count_calibration
```
Comment (local-minimum) [inline scanomatic/models/scanning_model.py]: above

File: scanomatic/ui_server/ui_server.py [hunk 6]
```diff
@@ -490,25 +490,26 @@ def _experiment():
...
+                description=data_object.get("description"),
+                email=data_object.get("email"),
+                pinning_formats=data_object.get("pinning_formats"),
+                fixture=data_object.get("fixture"),
+                scanner=data_object.get("scanner"),
+                scanner_hardware=data_object.get("scanner_hardware")
+                if "scanner_hardware" in request.json else "EPSON V700",
+                mode=data_object.get("mode", "TPU"),
+                plate_descriptions=plate_descriptions,
+                cell_count_calibration=data_object.get(
```
Comment (local-minimum) [inline scanomatic/ui_server/ui_server.py]: should be `id`

**Comment by skymandr** [comment]
This resolves #160.

**Pull request merged**